src/frontends/onnx/frontend/src/input_model.cpp

```diff
@@ -148,12 +149,69 @@ std::shared_ptr<Function> InputModelONNX::convert() {
 }
 
 // Editor features
+bool InputModelONNX::is_coorect_place(const ov::frontend::Place::Ptr& place) const {
+    if (const auto tensor = std::dynamic_pointer_cast<PlaceTensorONNX>(place)) {
+        return m_editor->is_correct_tensor_name(tensor->get_names()[0]);
+    } else if (const auto op = std::dynamic_pointer_cast<PlaceOpONNX>(place)) {
+        return m_editor->is_correct_and_unambiguous_node(op->get_editor_node());
+    } else if (const auto input_edge = std::dynamic_pointer_cast<PlaceInputEdgeONNX>(place)) {
+        if (auto tensor = std::dynamic_pointer_cast<PlaceInputEdgeONNX>(input_edge->get_source_tensor())) {
+            return m_editor->is_correct_tensor_name(tensor->get_names()[0]);
+        }
+    } else if (const auto output_edge = std::dynamic_pointer_cast<PlaceOutputEdgeONNX>(place)) {
+        if (auto tensor = std::dynamic_pointer_cast<PlaceInputEdgeONNX>(output_edge->get_target_tensor())) {
+            return m_editor->is_correct_tensor_name(tensor->get_names()[0]);
+        }
+    }
+    return false;
+}
+
+std::string InputModelONNX::get_place_name(const ov::frontend::Place::Ptr& place) const {
+    if (const auto tensor = std::dynamic_pointer_cast<PlaceTensorONNX>(place)) {
+        return place->get_names()[0];
+    } else if (const auto op = std::dynamic_pointer_cast<PlaceOpONNX>(place)) {
+        return op->get_editor_node().m_node_name;
+    } else if (const auto input_edge = std::dynamic_pointer_cast<PlaceInputEdgeONNX>(place)) {
+        if (auto tensor = std::dynamic_pointer_cast<PlaceInputEdgeONNX>(input_edge->get_source_tensor())) {
+            return tensor->get_names()[0];
+        }
+    } else if (const auto output_edge = std::dynamic_pointer_cast<PlaceOutputEdgeONNX>(place)) {
+        if (auto tensor = std::dynamic_pointer_cast<PlaceInputEdgeONNX>(output_edge->get_target_tensor())) {
+            return tensor->get_names()[0];
+        }
+    }
+    throw std::invalid_argument("Incorrect place type");
+}
+
 void InputModelONNX::override_all_outputs(const std::vector<Place::Ptr>& outputs) {
```

Comment:
Let's try to do sth like:
```
void InputModel::override_all_outputs(const std::vector<Place::Ptr>& outputs) {

    std::vector<Place::Ptr> expected_valid_outputs;
    for (const auto& output : expected_valid_outputs) {
        bool is_correct = is_coorect_place(output);
        if (!is_correct)
            NGRAPH_WARN << "Name  " << output->get_names().at(0),
                        << " of output node is not a correct node name. Ignoring this parameter.";
        else
            expected_valid_outputs.push_back(output);
    }

    extract_subgraph({}, expected_valid_outputs);

    NGRAPH_CHECK(std::all_of(std::begin(valid_outputs),
                             std::end(valid_outputs),
                             [](const Place::Ptr& place) {
                                 return place->is_output();
                             }),
                 "Not all provided arguments of override_all_outputs are new outputs of the model");

    const auto current_outputs = get_outputs();
    NGRAPH_CHECK(std::all_of(std::begin(current_outputs),
                             std::end(current_outputs),
                            [](const Place::Ptr& current_out) {
                                return std::find_if(std::begin(expected_valid_outputs), std::end(expected_valid_outputs)
                                [&current_out](const Place::Ptr& expected_out) {
                                    expected_out->is_equal(current_out);
                                }) != std::end(current_outputs);
                            }),
                "Some other than expected outputs were created during override_all_outputs");
}
```

Reply:
Updated the code to use your suggestion.